Source/WebGPU/WebGPU/CommandEncoder.swift

```diff
@@ -32,7 +37,314 @@ public func resolveQuerySet(commandEncoder: WebGPU.CommandEncoder, querySet: Web
+                    return
+                }
+
+                let destinationOrigin = MTLOriginMake(Int(destination.origin.x), Int(destination.origin.y), 0);
+                for layer in 0...copySize.depthOrArrayLayers {
+                    var layerTimesSourceBytesPerImage = UInt(layer)
+                    (layerTimesSourceBytesPerImage, didOverflow) = layerTimesSourceBytesPerImage.addingReportingOverflow(sourceBytesPerImage)
+                    guard !didOverflow else {
+                        return
+                    }
+                    var sourceOffset = UInt(source.layout.offset)
+                    (sourceOffset, didOverflow) = sourceOffset.addingReportingOverflow(layerTimesSourceBytesPerImage)
+                    guard !didOverflow else {
+                        return
+                    }
+                    var destinationSlice = UInt(destination.origin.z)
+                    (destinationSlice, didOverflow) = destinationSlice.addingReportingOverflow(UInt(layer))
+                    guard !didOverflow else {
+                        return
+                    }
+                    blitCommandEncoder.copy(
+                        from: sourceBuffer,
+                        sourceOffset: Int(sourceOffset),
+                        sourceBytesPerRow: Int(sourceBytesPerRow),
+                        sourceBytesPerImage: Int(sourceBytesPerImage),
+                        sourceSize: sourceSize,
+                        to: mtlDestinationTexture!,
+                        destinationSlice: Int(destinationSlice),
+                        destinationLevel: Int(destination.mipLevel),
+                        destinationOrigin: destinationOrigin,
+                        options: options
+                    )
+
+                }
+
+            case WGPUTextureDimension_3D.rawValue:
+                let sourceSize = MTLSizeMake(Int(widthForMetal), Int(heightForMetal), Int(depthForMetal));
+                guard widthForMetal != 0 && heightForMetal != 0 && depthForMetal != 0 else {
+                    return
+                }
+
+                let destinationOrigin = MTLOriginMake(Int(destination.origin.x), Int(destination.origin.y), Int(destination.origin.z))
+                let sourceOffset = UInt(source.layout.offset)
+                blitCommandEncoder.copy(
+                    from: sourceBuffer,
+                    sourceOffset: Int(sourceOffset),
+                    sourceBytesPerRow: Int(sourceBytesPerRow),
+                    sourceBytesPerImage: Int(sourceBytesPerImage),
+                    sourceSize: sourceSize,
+                    to: mtlDestinationTexture!,
+                    destinationSlice: 0,
+                    destinationLevel: Int(destination.mipLevel),
+                    destinationOrigin: destinationOrigin,
+                    options: options
+                )
+            case WGPUTextureDimension_Force32.rawValue:
+                // FIXME: need debug assert here.
```

Comment:
I don't think we need an assertion as this is validated earlier, `return` is fine

Reply:
Sure. Should I remove it from the ObjectiveC version as well then ? 
I honestly don't know what this code is doing so I am trying my best to convert it apples for apples.

Reply:
I'd suggest matching the ObjC code. assertionFailure() should do the trick.

Reply:
adding assertionFailure.